### DISCUSSION TITLE
disconnect `reason_code` meaning

### DISCUSSION THREAD

**Discussion by arturv2000** [nanomq/nanomq#1797] [Q&A]
Hi

When subscribing to `$SYS/brokers/disconnect` we receive a new message when a client disconnects

```
{
  "username": "admin",
  "ts": 1716560620524,
  "reason_code": "7",
  "client_id": "test1",
  "IPv4": "xxx.xxx.xxx.xxx"
}
```

What is the meaning for the field `reason_code`, I am getting a lot of values `7` and `6` and I am curious what the meaning, if it can indicate some kind of issue.

**NOTE:** Connections are being made using MQTT V3.1.1.

Thank You

**Comment by StargazerWayne** [comment]
You can refer to [MQTT spec](http://docs.oasis-open.org/mqtt/mqtt/v5.0/csprd02/mqtt-v5.0-csprd02.html#_Toc498345316) for `reason_code`. `6` or `7` is not a common one, can you provide more details to reproduce?

**Comment by JaylinYu** [comment]
6 or 7 probably due to keepalive timeout, they are NNG err_code
